index.js

```diff
@@ -59,3 +77,5 @@ module.exports = function requireAll(options) {
 function identity(val) {
   return val;
 }
+
+function isObject(arg) { return Object.prototype.toString.call(arg).indexOf('Object') !== -1; }
```

Comment:
Well, that's the question: I don't understand those extra unusual cases this one covers, so I don't know. My thought right now is that if I get a PR were someone says this is bad and changes it, I wouldn't even know what to argue back :) So I would at least need to understand the extra cases (or ideally they would be covered in our test suite).